sematic/runners/state_machine_runner.py

```diff
@@ -253,9 +264,25 @@ def _get_seconds_to_next_timeout(
 
     def _cancel_non_terminal_futures(self):
         for future in self._futures:
-            if not future.state.is_terminal():
+            state = self._read_refreshed_state(future)
+            if state != future.state:
```

Comment:
Please mention in the `_read_refreshed_state` docstring that implementations must not update `future.state` with the refreshed value.

Reply:
done